packages/x/vanilla/src/dom.ts

```diff
@@ -1,57 +1,77 @@
+/**
+  * TODO:
+  *  - DynamicFragment
+  *  - Namespaces
+  *  - SVG
+  *  - Modifier
+  *  - Portal
+  *  - SSR
+  *
+  *  Goals:
+  *   - Implement Glimmer compatibility
+  *   - Write compiler Glimmer -> whatever this DSL ends up being
+  *
+  * Stretch Goals:
+  *   - other compilers (html``)
+  */
 import type { Description, Reactive } from "@starbeam/interfaces";
 import { CachedFormula, DEBUG, type FormulaFn } from "@starbeam/reactive";
-import { RUNTIME } from "@starbeam/runtime";
+import { Resource,type ResourceBlueprint,RUNTIME,use } from "@starbeam/universal";
 
 import { Cursor } from "./cursor.js";
 
 export function Text(
   text: Reactive<string>,
   description?: string | Description
 ): ContentNode {
-  return ContentNode(({ into }) => {
-    const node = into.insert(into.document.createTextNode(text.read()));
+  return Content(
+    ({ into, owner }) => {
+      const node = into.insert(into.document.createTextNode(text.read()));
+      
+      return Resource(({ on }) => {
+        on.cleanup(() =>  void node.remove());
+        node.textContent = text.read();
+    })
+  }
 
-    return {
-      cleanup: () => void node.remove(),
-      update: () => (node.textContent = text.read()),
-    };
-  }, DEBUG?.Desc("resource", description, "Text"));
+  , DEBUG?.Desc('resource', description, 'Text'));
 }
 
 export function Comment(
   text: Reactive<string>,
   description?: string | Description
 ): ContentNode {
-  return ContentNode(({ into }) => {
```

Comment:
I renamed this to just `Content`, because the type overload was breaking my brain, especially since the two ContentType's weren't exactly referring to the same function